src/detectors/BEMC/ProtoCluster_factory_EcalBarrelSciGlassProtoClusters.h

```diff
@@ -37,7 +37,7 @@ class ProtoCluster_factory_EcalBarrelSciGlassProtoClusters : public JFactoryT<ed
         // Magic constants:
         //  24 - number of sectors
         //  5  - number of towers per sector
-        u_adjacencyMatrix = "(abs(tower_1 - tower_2) + (abs((sector_1 - sector_2) * 5 + row_1 - row_2) == 1) + (abs((sector_1 - sector_2) * 5 + row_1 - row_2) == (24 * 5 - 1))) == 1";
+        u_adjacencyMatrix = "(abs(tower_1 - tower_2) + min(abs((sector_1 - sector_2) * 5 + row_1 - row_2), 24 * 5 - abs((sector_1 - sector_2) * 5 + row_1 - row_2))) == 1";
         std::remove(u_adjacencyMatrix.begin(), u_adjacencyMatrix.end(), ' ');
```

Comment:
For my own mental parser, this should strip out newlines too...
```suggestion
        u_adjacencyMatrix = "
        (
          abs(tower_1 - tower_2)
        + min(
              abs((sector_1 - sector_2) * 5 + row_1 - row_2),
              24 * 5 - abs((sector_1 - sector_2) * 5 + row_1 - row_2)
          )
        ) == 1";
        u_adjacencyMatrix.erase(
          std::remove_if(u_adjacencyMatrix.begin(), u_adjacencyMatrix.end(), ::isspace),
          u_adjacencyMatrix.end());
```

Is that the same as:
```
        u_adjacencyMatrix = "
        ( 
          abs(tower_1 - tower_2)
        + abs(((sector_1 - sector_2) * 5 + row_1 - row_2) % (5*24))
        ) == 1";
```

Reply:
No, it's not. The idea of this is that for each two hits, a cylinder has two geodesic curves: one wraps around without crossing from tower 119 to 0 and one that does. We are just picking a shortest distance.

Reply:
Basically it's a taxicab metric on a cylinder.